PermanentOrg/web-app
# PER-8337: Public Download

## Description
This PR adds a download button to items on public archives. It also removes the PublicItemComponent, which seemed to be unused as items in public galleries instead use the normal FileViewerComponent instead.

While going through other public code, I'm wondering if the `SearchComponent` (not to be confused with the `SearchBoxComponent`) can be removed from the codebase as well. It seems like it was just used to test functionality when the search box was being developed and was just kept in the codebase all this time. This component is actually connected to a route so you can view it at `/p/search`. [You can even view it on production.](https://permanent.org/p/search)

Resolves PER-8337.

## Steps to Test
- Go to a public gallery in your local instance.
- Click on an item to view it in "full view" mode.
- Click the download button and ensure the file downloads properly.
- *Optional:* Navigate to `/p/search` in your local instance and look at the page. There's nothing to test here but... it's weird, right?
